packages/node/src/integrations/spotlight.ts

```diff
@@ -0,0 +1,110 @@
+import type { Client, Integration } from '@sentry/types';
+import { logger, serializeEnvelope } from '@sentry/utils';
+import * as http from 'http';
+import { URL } from 'url';
+
+type SpotlightConnectionOptions = {
+  /**
+   * Set this if the Spotlight Sidecar is not running on localhost:8969
+   * By default, the Url is set to http://localhost:8969
+   */
+  sidecarUrl?: string;
+};
+
+/**
+ * Use this integration to send errors and transactions to Spotlight.
+ *
+ * Learn more about spotlight at https://spotlightjs.com
+ *
+ * Important: This integration only works with Node 18 or newer
+ */
+export class Spotlight implements Integration {
+  public static id = 'Spotlight';
+  public name = Spotlight.id;
+
+  private readonly _options: Required<SpotlightConnectionOptions>;
+
+  public constructor(options?: SpotlightConnectionOptions) {
+    this._options = {
+      sidecarUrl: options?.sidecarUrl || 'http://localhost:8969',
+    };
+  }
+
+  /**
+   * JSDoc
+   */
+  public setupOnce(): void {
+    // empty but otherwise TS complains
+  }
+
+  /**
+   * Sets up forwarding envelopes to the Spotlight Sidecar
+   */
+  public setup(client: Client): void {
+    if (process.env.NODE_ENV !== 'development') {
+      logger.warn("[Spotlight] It seems you're not in dev mode. Do you really want to have Spoltight enabled?");
+    }
+    connectToSpotlight(client, this._options);
+  }
+}
+
+function connectToSpotlight(client: Client, options: Required<SpotlightConnectionOptions>): void {
+  const spotlightUrl = parseSidecarUrl(options.sidecarUrl);
+  if (!spotlightUrl) {
+    return;
+  }
+
+  let failedRequests = 0;
+
+  if (typeof client.on !== 'function') {
+    logger.warn('[Spotlight] Cannot connect to spotlight due to missing method on SDK client (`client.on`)');
+    return;
+  }
+
+  client.on('beforeEnvelope', envelope => {
+    if (failedRequests > 3) {
+      logger.warn('[Spotlight] Disabled Sentry -> Spotlight integration due to too many failed requests');
+      return;
+    }
+
+    const serializedEnvelope = serializeEnvelope(envelope);
+
+    const req = http.request(
+      {
+        method: 'POST',
+        path: '/stream',
```

Comment:
Looks like `spotlightUrl.path` should be `/stream` already? Is hardcoding intentional?

Reply:
If I'm not mistaken (it's laaate here 😅) `spotlightUrl` is the optionally user configurable part of the URL - protocol, host and port. The path I think we're going to mandate to be `/stream`.

Reply:
Id just make it the full URL tbqh. That makes it easier for us to swap out different proxies later if we wanted.

e.g. make it http://localhost:8969/stream as the default (or w/e it is)